discovery_engine/lib/src/domain/engine/engine.dart

```diff
@@ -12,13 +12,33 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import 'dart:typed_data' show Uint8List;
+
 import 'package:xayn_discovery_engine/src/domain/models/active_data.dart'
     show ActiveDocumentData;
 import 'package:xayn_discovery_engine/src/domain/models/document.dart'
-    show Document;
+    show Document, DocumentFeedback;
+import 'package:xayn_discovery_engine/src/domain/models/unique_id.dart'
+    show DocumentId, StackId;
 
 /// Interface to Discovery Engine core.
 abstract class Engine {
   /// Retrieves at most [maxDocuments] feed documents.
   Map<Document, ActiveDocumentData> getFeedDocuments(int maxDocuments);
+
+  /// Process the feedback about the user spending some time on a document.
+  void timeLogged(
+    DocumentId docId, {
+    required Uint8List smbertEmbedding,
+    required Duration seconds,
+  });
+
+  /// Process the feedback about the user reacting to a document.
+  void userReacted(
+    DocumentId docId, {
+    required StackId stackId,
+    required String snippet,
```

Comment:
snippet was added cause @Robert-Steiner is doing same in the Rust counterpart

Reply:
#72 has been merged